src/store_client.cc

```diff
@@ -622,28 +668,34 @@ store_client::readHeader(char const *buf, ssize_t len)
     if (len < 0)
         return fail();
 
-    if (!unpackHeader(buf, len)) {
-        fail();
-        return;
-    }
+    MemObject *const mem = entry->mem_obj;
+    const auto initialRead = (mem->swap_hdr_sz == 0);
```

Comment:
The old readHeader() did not, by itself, parse HTTP header. The new readHeader() does not, by itself, parse HTTP header. In both cases, HTTP header parsing happens in readBody(). Yes, readHeader() calls readBody() but that is just how the state machine is making progress in this imperfect code. A better implementation would move that readBody() call from readHeader() into a new dedicated method (that fileRead() would schedule). That improvement may be outside this PR scope (that remains to be seen), but, in any case, we should keep in mind that the original code had this divide -- readHeader() is for reading the swapout header and readBody() is for reading everything else (including HTTP headers).

I suspect we can and probably should preserve the original readHeader()/readBody() roles while fixing HTTP header loading. For now, I will just ask one question: For the sake of the argument, imagine that readBody() never throws and, hence, we do not need to catch its exceptions. Instead of throwing, our readBody() will handle insufficient data internally. For example, you can imagine that readBody() became a readBodyOrThrow() wrapper that catches exceptions, and that the current readBody() code was moved into readBodyOrThrow(). Can we leave readHead() and readFile() unchanged in this case?

```C++
readBody() {
    try {
        readBodyOrThrow();
        ...
    } catch ...
}
```

Reply:
>  we can and probably should preserve the original readHeader()/readBody() roles 

Done (at 714614b). I also removed the 'insufficient data' exception throwing: readBody() now itself initiates a new fileRead().